tests/integration/test_relation.py

```diff
@@ -0,0 +1,136 @@
+import asyncio
+import json
+from pathlib import Path
+
+import kubernetes
+import requests
+from pytest_operator.plugin import OpsTest
+
+INGRESS_APP_NAME = "ingress"
+ANY_APP_NAME = "any"
+NEW_HOSTNAME = "any.other"
+NEW_PORT = 18080
+
+
+async def test_build_and_deploy(ops_test: OpsTest, run_action):
+    """
+    arrange: no arrange.
+    act: build and deploy nginx-ingress-integrator charm, also deploy and relate an any-charm
+        application for test purposes.
+    assert: all application should be active.
+    """
+    ingress_lib = Path("lib/charms/nginx_ingress_integrator/v0/ingress.py").read_text()
+    any_charm_script = Path("tests/integration/any_charm.py").read_text()
+    any_charm_src_overwrite = {
+        "ingress.py": ingress_lib,
+        "any_charm.py": any_charm_script,
+    }
+
+    async def build_and_deploy():
+        charm = await ops_test.build_charm(".")
+        return await ops_test.model.deploy(
+            str(charm), application_name="ingress", series="focal", trust=True
+        )
+
+    await asyncio.gather(
+        ops_test.model.deploy(
+            "any-charm",
+            application_name="any",
+            channel="beta",
+            config={"src-overwrite": json.dumps(any_charm_src_overwrite)},
+        ),
+        build_and_deploy(),
+    )
+    await ops_test.model.wait_for_idle()
+    await run_action(ANY_APP_NAME, "rpc", method="start_server")
+    await ops_test.model.add_relation(ANY_APP_NAME, f"{INGRESS_APP_NAME}:ingress")
+    await ops_test.model.wait_for_idle()
+
+
+async def test_ingress_connectivity():
```

Comment:
Should this depend on a fixture that builds and deploys the charm?

Reply:
Same as above.